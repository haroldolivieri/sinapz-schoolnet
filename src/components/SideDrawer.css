.drawer {
    position: relative;
    z-index: 9999;
    height: 100vh;
    width: 80vw;
    background-color: #3C4754;
}

@media only screen and (min-width: 768px) {
    .drawer { 
        width: 30vw;   
    }
}