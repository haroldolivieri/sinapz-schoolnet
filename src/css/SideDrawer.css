.drawer {
    display: flex;
    flex-direction: column;
    align-items: center;
    position: relative;
    z-index: 9999;
    height: 100vh;
    width: 70vw;
    background-color: #3C4754;
}
@media only screen and (min-width: 900px) {
    .drawer { 
        width: 20vw;   
    }
}

.avatar {
    border-radius: 50%;
    width:8em;
    height:8em;
    margin: 1.5em;
}

.title {
    color: white;
    font-size: 1.2em;
    font-weight: 600;
    margin-bottom: 1em;
}

.list {
    width: 100%;
}

.item-text {
    font-size: 0.8em;
    color: white;
}