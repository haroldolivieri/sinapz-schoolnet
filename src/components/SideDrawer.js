import React from 'react'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import PropTypes from 'prop-types'
import { compose } from 'redux'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import '../css/SideDrawer.css'

export const DrawerItem = ({
    text = 'Unknown',
    icon = <ItemIcon />,
    action = function() {}
}) => (
    <ListItem id="list-item" button key={text} onClick={action}>
        <ListItemIcon>{icon}</ListItemIcon>
        <p className="item-text">{text}</p>
    </ListItem>
)
DrawerItem.propTypes = {
    text: PropTypes.string,
    icon: PropTypes.element,
    action: PropTypes.func
}

export const ItemIcon = ({ img = 'undefined.svg', alt = '' }) => (
    <img src={require(`../images/${img}`)} alt={alt} />
)

export const SideDrawer = ({
    history,
    menuItems = [],
    schoolnetInfo = {
        name: 'Anonymous',
        photo:
            'https://cdn.shopify.com/s/files/1/0539/4361/products/anonymous-guy-fawkes-anarchy-decal.jpg?v=1410346765'
    }
}) => {
    const handleItemSelected = pathTo => {
        history.push(pathTo)
    }

    const AvatarImage = ({ url, alt = '' }) => (
        <img src={url} className="avatar" alt={alt} />
    )

    return (
        <div className="drawer">
            <AvatarImage url={schoolnetInfo.photo} alt={schoolnetInfo.name} />
            <span className="title">{schoolnetInfo.name}</span>
            <div className="list">
                {menuItems.map(item => (
                    <DrawerItem
                        id="drawer-item"
                        key={item.name}
                        text={item.name}
                        icon={<ItemIcon img={item.icon} alt={item.name} />}
                        action={handleItemSelected.bind(this, item.path)}
                    />
                ))}
            </div>
        </div>
    )
}

const mapStateToProps = state => ({
    schoolnetInfo: state.menu.get('schoolnetInfo'),
    menuItems: state.menu.get('menuItems')
})

export default compose(
    withRouter,
    connect(mapStateToProps)
)(SideDrawer)

SideDrawer.propTypes = {
    history: PropTypes.object,
    menuItems: PropTypes.array,
    schoolnetInfo: PropTypes.object
}
