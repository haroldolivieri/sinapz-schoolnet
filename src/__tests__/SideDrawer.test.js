import React from 'react';
import SideDrawer from '../components/SideDrawer';
import { shallow } from 'enzyme';

describe('Side Drawer', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<SideDrawer/>)
        expect(wrapper).toMatchSnapshot()
    });
});
