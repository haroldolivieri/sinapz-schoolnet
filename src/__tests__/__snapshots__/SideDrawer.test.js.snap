// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Drawer renders correctly 1`] = `
<div
  className="drawer"
>
  <AvatarImage
    url="https://cdn.shopify.com/s/files/1/0539/4361/products/anonymous-guy-fawkes-anarchy-decal.jpg?v=1410346765"
  />
  <WithStyles(Typography)
    className="title"
  >
    Escola Anônima
  </WithStyles(Typography)>
  <div
    className="list"
  >
    <DrawerItem
      text="Dashboard"
    />
    <DrawerItem
      text="Agenda"
    />
    <DrawerItem
      text="Filiais"
    />
    <DrawerItem
      text="Professores"
    />
    <DrawerItem
      text="Variáveis da prova"
    />
    <DrawerItem
      text="Matérias disponíveis"
    />
  </div>
</div>
`;
